source/choose-connection-target.txt

```diff
@@ -0,0 +1,173 @@
+- Specify more than one host in the replica set.
+
+In the following example, the driver uses a sample connection URI to connect to the 
+MongoDB replica set ``sampleRS``, which is running on port ``27017`` of three different
+hosts, including ``host1``:
+
+.. code-block:: java
+
+    import com.mongodb.reactivestreams.client.MongoClients;
+    import com.mongodb.reactivestreams.client.MongoClient;
+
+    public class MongoConnection {
+        public static void main(String[] args) {
+            String uri = "mongodb://localhost:27017/?replicaSet=sampleRS";
+            try (MongoClient client = MongoClients.create(uri)) {
+                // use `client` here
+            }
+        }
+    }
+
+.. note::
+
+   The method constructing a ``MongoClient`` is *non-blocking*.
+   When you connect to a replica set, the method returns immediately while the
+   client uses background threads to connect to the replica set.
+   
+   If you create a ``MongoClient`` and immediately print the string representation
+   of its ``nodes`` attribute, the list might be empty while the client connects to
+   the replica-set members.
+
+Initialization
+~~~~~~~~~~~~~~
+
+To initialize a replica set, you must connect directly to a single member. To do so,
+set the ``directConnection`` connection
+option to ``True``. You can do this in two ways: by passing an argument to the
+``MongoClient`` constructor or through a parameter in your connection string.
+
+.. tabs::
+
+   .. tab:: MongoClient
+      :tabid: mongoclient
+
+      .. code-block:: java
+         :emphasize-lines: 7
+
+         import com.mongodb.reactivestreams.client.MongoClients;
+         import com.mongodb.reactivestreams.client.MongoClient;
+
+         public class MongoConnection {
+            public static void main(String[] args) {
+                try (MongoClient client = MongoClients.create("mongodb://<hostname>:<port>",
+                directConnection=True)){
+                    // use `client` here
+                }
+            }
+         }
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. code-block:: java
+         :emphasize-lines: 6
+
+         import com.mongodb.reactivestreams.client.MongoClients;
+         import com.mongodb.reactivestreams.client.MongoClient;
+
+         public class MongoConnection {
+            public static void main(String[] args) {
+                String uri = "mongodb://<hostname>:<port>/?directConnection=true";
+                try (MongoClient client = MongoClients.create(uri)){
+                    // use `client` here
+                }
+            }
+         }
+
+API Documentation
+-----------------
+
+To learn more about creating a ``MongoClient`` instance in the {+driver-short+},
+see the following API documentation:
+
+- `MongoClient <https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-reactivestreams/com/mongodb/reactivestreams/client/MongoClients.html>`__ 
```

Comment:
S: instead of pasting the whole string, use a source constant that you can reuse. Lmk if you have questions about how to do this